compiler/src/main/scala/org/scalajs/core/compiler/JSEncoding.scala

```diff
@@ -61,7 +61,7 @@ trait JSEncoding extends SubComponent { self: GenJSCode =>
       longName = base+"$"+suffix
     }
     usedLocalNames += longName
-    longName
+    mangleJSName(longName)
```

Comment:
Oh... Didn't even realize the reserved... This is actually a bug in `mangleJSName`, no? We should probably file this for 1.0.

In this case, maybe better we don't change this...